pkg/sdk/poc/generator/templates.go

```diff
@@ -3,186 +3,289 @@ package generator
-{{define "DTO_STRUCT"}}
-type {{.DtoDecl}} struct {
-	{{- range .Fields}}
-		{{- if .ShouldBeInDto}}
-		{{.Name}} {{.DtoKind}} {{if .Required}}// required{{end}}
-		{{- end}}
-	{{- end}}
-}
-
-{{- range .Fields}}
-	{{if .IsStruct}}
-	{{template "DTO_STRUCT" .}}
-	{{end}}
-{{- end}}
-{{end}}
-
-//go:generate go run ../../dto-builder-generator/main.go
+//go:generate go run ./dto-builder-generator/main.go
 
 var (
-	{{- range .Operations}}
-	_ optionsProvider[{{.OptsField.KindNoPtr}}] = new({{.OptsField.DtoDecl}})
-	{{- end}}
+	{{- range .Operations }}
+	_ optionsProvider[{{ .OptsField.KindNoPtr }}] = new({{ .OptsField.DtoDecl }})
+	{{- end }}
 )
+`)
 
-{{- range .Operations}}
-	{{template "DTO_STRUCT" .OptsField}}
-{{- end}}
+var DtoDeclTemplate, _ = template.New("dtoTemplate").Parse(`
+type {{ .DtoDecl }} struct {
+	{{- range .Fields }}
+		{{- if .ShouldBeInDto }}
+		{{ .Name }} {{ .DtoKind }} {{ if .Required }}// required{{ end }}
+		{{- end }}
+	{{- end }}
+}
 `)
 
-var ImplementationTemplate, _ = template.New("implementationTemplate").Parse(`
-{{define "MAPPING" -}}
-	&{{.KindNoPtr}}{
-		{{- range .Fields}}
-			{{- if .ShouldBeInDto}}
-			{{if .IsStruct}}{{else}}{{.Name}}: r{{.Path}},{{end}}
-			{{- end}}
-		{{- end}}
+var ImplementationTemplate, _ = template.New("implementationTemplate").
+	Funcs(template.FuncMap{
+		"deref": func(p *DescriptionMappingKind) string { return string(*p) },
+	}).
+	Parse(`
+{{ define "MAPPING" -}}
+	&{{ .KindNoPtr }}{
+		{{- range .Fields }}
+			{{- if .ShouldBeInDto }}
+			{{ if .IsStruct }}{{ else }}{{ .Name }}: r{{ .Path }},{{ end -}}
+			{{- end -}}
+		{{- end }}
 	}
-	{{range .Fields}}
-		{{if .ShouldBeInDto}}
-			{{if .IsStruct}}
-				if r{{.Path}} != nil {
-					opts{{.Path}} = {{template "MAPPING" .}}
+	{{- range .Fields }}
+		{{- if .ShouldBeInDto }}
+			{{- if .IsStruct }}
+				if r{{ .Path }} != nil {
+					{{- if not .IsSlice }}
+						opts{{ .Path }} = {{ template "MAPPING" . -}}
+					{{- else }}
+						s := make({{ .Kind }}, len(r{{ .Path }}))
+						for i, v := range r{{ .Path }} {
+							s[i] = {{ .KindNoSlice }}{
+							  {{- range .Fields }}
+								   {{ .Name }}: v.{{ .Name }},
+							  {{- end }}
+							}
+						}
+						opts{{ .Path }} = s
+					{{ end -}}
 				}
-			{{end}}
-		{{end}}
-	{{end}}
-{{end}}
+			{{- end -}}
+		{{ end -}}
+	{{ end }}
+{{ end }}
+{{ define "MAPPING_FUNC" }}
+	func (r {{ .From.Name }}) {{ .MappingFuncName }}() *{{ .To.KindNoPtr }} {
+		// TODO: Mapping
+		return &{{ .To.KindNoPtr }}{}
+	}
+{{ end }}
 import "context"
 
-{{$impl := .NameLowerCased}}
-var _ {{.Name}} = (*{{$impl}})(nil)
+{{ $impl := .NameLowerCased }}
+var _ {{ .Name }} = (*{{ $impl }})(nil)
 
-type {{$impl}} struct {
+type {{ $impl }} struct {
 	client *Client
 }
-{{range .Operations}}
-func (v *{{$impl}}) {{.Name}}(ctx context.Context, request *{{.OptsField.DtoDecl}}) error {
-	opts := request.toOpts()
-	return validateAndExec(v.client, ctx, opts)
-}
-{{end}}
+{{ range .Operations }}
```

Comment:
this have to be cleaned up, please add to essentials (I know there is a template separation one, but also a rework here would be nice to make it much simpler template)